src/NUnitFramework/framework/DatabaseData.cs

```diff
@@ -0,0 +1,382 @@
+        /// <returns></returns>
+        IEnumerable Query(string qry);
+
+
+        /// <summary>
+        /// The name of the driver as specified in the connection string
+        /// </summary>
+        string DatabaseDriverName { get; }
+    }
+
+    /// <summary>
+    /// Maps the various database vendors to their appropriate .NET driver
+    /// </summary>
+    public class DatabaseConnectionType
+    {
+
+        /// <summary>
+        /// The mapping of the various database types to the local database driver in .NET
+        /// </summary>
+        IList<string> drivers = null;
+
+        /// <summary>
+        /// The numeric ID of the driver type
+        /// </summary>
+        public int Id { get; private set; }
+
+        /// <summary>
+        /// The database driver type
+        /// </summary>
+        public string Name { get; private set; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="id">Identification of symbol</param>
+        /// <param name="driverName">Name of the driver</param>
+        /// <param name="dbNames">List of string drivers used for mapping different driver types.</param>
+        private DatabaseConnectionType(int id, string driverName, IList<string> dbNames)
+        {
+            Id = id;
+            Name = driverName;
+            drivers = dbNames;
+        }
+
+        /// <summary>
+        /// No type specified 
+        /// </summary>
+        public static DatabaseConnectionType None = new DatabaseConnectionType(-1, "None", null);
+
+        /// <summary>
+        /// Generic .NET DataProvider (SqlClient)
+        /// </summary>
+        public static DatabaseConnectionType Generic = new DatabaseConnectionType(0, "Generic", new List<string>() { "SQLExpress" });
+
+        /// <summary>
+        /// ODBC connection
+        /// </summary>
+        public static DatabaseConnectionType ODBC = new DatabaseConnectionType(1, "ODBC", new List<string>() { "SQL Server", "Microsoft Access Driver", "Microsoft ODBC Driver for Oracle", "Microsoft ODBC for Oracle", "Oracle ODBC Driver", "IBM DB2 ODBC DRIVER", "MySql", "MySql ODBC 3.51 Driver", "Microsoft Excel Driver", "Microsoft Text Driver" });
+
+        /// <summary>
+        /// OLE connection
+        /// </summary>
+        public static DatabaseConnectionType OLE = new DatabaseConnectionType(2, "OLE", new List<string>() { "SQLOLEDB", "MSDAORA", "DB2OLEDB", "MySqlProv" });
+
+        /// <summary>
+        /// Checks a connection string for the appropriate driver type
+        /// </summary>
+        /// <param name="connString"></param>
+        /// <returns>Tuple of [bool,string], one verifying the driver type, the second the name of the specific driver</returns>
+        public Tuple<bool, string> Verify(string connString)
+        {
+            if (drivers == null)
+                return new Tuple<bool, string>(false, string.Empty);
+
+            foreach (var dvr in drivers)
```

Comment:
I only wanted to do the drivers that are bundled with the framework as a first iteration.